spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/support/SimpleReactiveMongoRepository.java

```diff
@@ -404,6 +405,14 @@ public SimpleReactiveMongoRepository(MongoEntityInformation<T, ID> entityInforma
 		return Flux.fromIterable(entities).flatMap(this::delete).then();
 	}
 
+	@Override
+	public Mono<Void> deleteAllById(Iterable<? extends ID> ids) {
+
+		Assert.notNull(ids, "The given Iterable of ids must not be null!");
+
+		return Flux.fromIterable(ids).flatMap(this::deleteById).then();
```

Comment:
We should collect all ID's into a query to run a single delete query and to avoid issuing a query per given Id.

Reply:
Should we do the same for `public Mono<Void> deleteAll(Iterable<? extends T> entities)`?

Reply:
I made the same change for `public Mono<Void> deleteAll(Iterable<? extends T> entities)` as well in a separate commit. 

Pick and choose.